nl/nl_linux.go

```diff
@@ -440,6 +440,11 @@ done:
 				continue
 			}
 			if m.Header.Type == unix.NLMSG_DONE {
+				native := NativeEndian()
+				error := int32(native.Uint32(m.Data[0:4]))
+				if error < 0 {
+					return nil, syscall.Errno(-error)
+				}
```

Comment:
Maybe this if condition and the next case (`unix.NLMSG_ERROR`) can completely share the implementation by just expanding the if condition below on line 450 to include the `unix.NLMSG_DONE` case?